cmd/collector/app/span_processor.go

```diff
@@ -123,10 +123,12 @@ func newSpanProcessor(spanWriter spanstore.Writer, additional []ProcessSpan, opt
 
 	processSpanFuncs := []ProcessSpan{options.preSave, sp.saveSpan}
```

Comment:
Couldn't we just do this? I don't think this requires any new cmd line options
```suggestion
	processSpanFuncs := []ProcessSpan{options.preSave, sp.saveSpan, sp.countSpan}
```

Reply:
Thank you for your suggestion!
From the discussions in https://github.com/jaegertracing/jaeger/issues/2126 , my understanding is that this metrics should be an optional due to performance concerns ( https://github.com/jaegertracing/jaeger/issues/2126#issuecomment-600038153 ).
If there is no problem in always enabling it, the code will be as suggested.

Reply:
oh, i see. Then I guess it's ok to have a flag, but I would still make the counting only dependent on that flag, not on the dynqueue

Reply:
Hmm, `bytesProcessed` is used by both metrics and dynqueue, so the counting is needed if either flag is true.

https://github.com/jaegertracing/jaeger/blob/9b4bbf4f75bac3333f1099e26cc7cf3dd79773bf/cmd/collector/app/span_processor.go#L269

How about to count in either case and use `--collector.enable-processed-span-metrics` flag to determine whether to update the metrics? This makes the role of the flag straightforward.

```go
func (sp *spanProcessor) updateGauges() {
        if sp.processedSpanMetricsEnabled {
                sp.metrics.SpansBytes.Update(int64(sp.bytesProcessed.Load()))
        }
```

Reply:
The bumping of the metric itself is very cheap, there's no need to protect it with a flag. 

There's certainly a number of places touching `bytesProcessed`. I am not particularly liking this design - couldn't this int64 variable overflow with time? But that's a problem separate from this PR.

The main concern from the ticket was the overhead of `span.Size()` calls.

Reply:
This performance concern is existing issue of dynqueue and it doesn't come up unless the flag is true, so why not discuss about performance and design on a separate ticket?

I think the following issues should be handled by separate PRs.
- issue that metrics cannot be collected unless dynqueue is enabled (resolved by this PR)
- performance tuning the counting
- improving design not to depend too much on bytesProcessed and to avoid risk of overflow

The latter two seem to require more discussion before code changing.